src/test/scala/net/snowflake/spark/snowflake/ConversionsSuite.scala

```diff
@@ -193,4 +193,88 @@ class ConversionsSuite extends FunSuite {
 
     assert(expect == result.toString())
   }
+
+  test("Data with micro-seconds and nano-seconds precision should be correctly converted"){
```

Comment:
This is a unit test. It proves that it parse timestamp with micro/nano-seconds.
Could you please add an integration test  (AKA end-to-end test)?